src/App.tsx

```diff
@@ -1,60 +1,187 @@
-import classNames from 'classnames';
+/* eslint-disable @typescript-eslint/indent */
 
+import classNames from 'classnames';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import { useEffect, useState } from 'react';
 
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { PostDetails } from './components/PostDetails';
+import { PostsList } from './components/PostsList';
 import { Loader } from './components/Loader';
+import { Comment, CommentData } from './types/Comment';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import {
+  addNewComment,
+  deleteComment,
+  getUserPostComments,
+  getUserPosts,
+  getUsers,
+} from './api/post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+  const [showPostLoader, setShowPostLoader] = useState<boolean>(false);
+  const [showCommentLoader, setShowCommentLoader] = useState<boolean>(false);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isSidebarVisible, setIsSidebarVisible] = useState<number | null>(null);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [hasPostError, setHasPostError] = useState<boolean>(false);
+  const [hasCommentError, setHasCommentError] = useState<boolean>(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(res => {
+        setUsers(res as User[]);
+      })
+      .catch(() => new Error('Failed to fetch users'));
```

Comment:
You should handle the error in catch block. It's better to set some state variable in case of error and show error message to the user.